src/wallet/sqlite.cpp

```diff
@@ -80,6 +81,72 @@ void SQLiteDatabase::Cleanup()
 
 void SQLiteDatabase::Open(const char* mode)
 {
+    int flags = SQLITE_OPEN_FULLMUTEX | SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE;
+    if (m_mock) {
+        flags |= SQLITE_OPEN_MEMORY; // In memory database for mock db
+    }
+
+    if (m_db == nullptr) {
+        TryCreateDirectories(m_dir_path);
+        int ret = sqlite3_open_v2(m_file_path.c_str(), &m_db, flags, nullptr);
+        if (ret != SQLITE_OK) {
+            throw std::runtime_error(strprintf("SQLiteDatabase: Failed to open database: %s\n", sqlite3_errstr(ret)));
+        }
+    }
+
```

Comment:
6720df53bd5d4e551c101743c4050bd0c7682238
`Assert(m_db);` ?

Reply:
I think that is unnecessary.